### PR TITLE
[Cart] change remove order item from PATCH method to DELETE method

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Controller/OrderController.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    private $commandBus;
+
+    public function __construct(MessageBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $command = new RemoveItemFromCart();
```
Comment (SirDomin) [inline src/Sylius/Bundle/ApiBundle/Controller/OrderController.php:34]:
```suggestion
        $command = new RemoveItemFromCart($request->attributes->get('id'), $request->attributes->get('itemId'));
```
wouldnt it be better?

File: src/Sylius/Bundle/ApiBundle/Controller/OrderController.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace Sylius\Bundle\ApiBundle\Controller;
+
+use Sylius\Bundle\ApiBundle\Command\Cart\RemoveItemFromCart;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class OrderController
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/Controller/OrderController.php:22]:
```suggestion
final class RemoveOrderItemAction
```
or 
```suggestion
final class DeleteOrderItemAction
```

File: src/Sylius/Bundle/ApiBundle/Command/Cart/RemoveItemFromCart.php [hunk 3]
```diff
@@ -50,4 +42,14 @@ public function setOrderTokenValue(?string $orderTokenValue): void
     {
         $this->orderTokenValue = $orderTokenValue;
     }
+
+    public function getItemId(): string
+    {
+        return $this->itemId;
+    }
+
+    public function setItemId(string $itemId): void
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Command/Cart/RemoveItemFromCart.php:56]: Do you need this function?

File: src/Sylius/Bundle/ApiBundle/CommandHandler/Cart/RemoveItemFromCartHandler.php [hunk 4]
```diff
@@ -24,33 +25,40 @@
...
     ) {
+        $this->orderRepository = $orderRepository;
         $this->orderItemRepository = $orderItemRepository;
         $this->orderModifier = $orderModifier;
     }
 
     public function __invoke(RemoveItemFromCart $removeItemFromCart): OrderInterface
     {
+        /** @var OrderInterface|null $cart */
+        $cart = $this->orderRepository->findOneBy(['tokenValue' => $removeItemFromCart->orderTokenValue]);
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/Cart/RemoveItemFromCartHandler.php:50]:
```suggestion
        $cart = $this->orderRepository->findCartByTokenValue($removeItemFromCart->orderTokenValue);
```

Comment (GSadee) [reply src/Sylius/Bundle/ApiBundle/CommandHandler/Cart/RemoveItemFromCartHandler.php:50]: And I'm not sure if this database call is needed if you are using also the `orderTokenValue` to find the order item below.

File: src/Sylius/Bundle/ApiBundle/Resources/config/services.xml [hunk 5]
```diff
@@ -127,5 +127,9 @@
             <argument type="service" id="sylius.repository.payment"/>
             <argument type="service" id="sylius.repository.payment_method"/>
         </service>
+
+        <service id="DeleteOrderItemAction" public="true">
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services.xml:131]: I would be for FQCN as a service id

File: src/Sylius/Bundle/ApiBundle/spec/CommandHandler/Cart/RemoveItemFromCartHandlerSpec.php [hunk 6]
```diff
@@ -16,8 +16,10 @@
 use PhpSpec\ObjectBehavior;
 use Sylius\Bundle\ApiBundle\Command\Cart\RemoveItemFromCart;
 use Sylius\Bundle\OrderBundle\Doctrine\ORM\OrderItemRepository;
+use Sylius\Component\Core\Model\Order;
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/CommandHandler/Cart/RemoveItemFromCartHandlerSpec.php:19]: Changes in this spec are not necessary I guess

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Adam! :tada:

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Order.xml [hunk 7]
```diff
@@ -268,15 +268,33 @@
             </itemOperation>
 
             <itemOperation name="shop_remove_item">
-                <attribute name="method">PATCH</attribute>
-                <attribute name="path">/shop/orders/{id}/remove</attribute>
+                <attribute name="method">DELETE</attribute>
+                <attribute name="path">/shop/orders/{id}/items/{itemId}</attribute>
                 <attribute name="messenger">input</attribute>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Order.xml:273]: probably not needed anymre

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Order.xml [hunk 8]
```diff
@@ -268,15 +268,33 @@
...
-                <attribute name="input">Sylius\Bundle\ApiBundle\Command\Cart\RemoveItemFromCart</attribute>
+                <attribute name="controller">Sylius\Bundle\ApiBundle\Controller\DeleteOrderItemAction</attribute>
+                <attribute name="write">false</attribute>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Order.xml:275]: Is it needed?

Comment (AdamKasp) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Order.xml:275]: yes, without it, after job from the controller, Api-platform will delete order (cart in this case).

**Pull request merged**